v5/pivotal/aggregator.go

```diff
@@ -0,0 +1,218 @@
+		service:            a,
+	}
+	return &aggregation
+}
+
+func newAggregatorService(client *Client) *AggregatorService {
+	return &AggregatorService{client}
+}
+
+// Story adds a story request to the aggregation.
+func (a *Aggregation) Story(projectID, storyID int) *Aggregation {
+	a.requests = append(a.requests, buildStoryURL(projectID, storyID))
+	return a
+}
+
+// StoryUsingStoryID adds a story request using only using the story ID.
+func (a *Aggregation) StoryUsingStoryID(storyID int) *Aggregation {
+	a.requests = append(a.requests, buildStoryURLOnlyUsingStoryID(storyID))
+	return a
+}
+
+// Stories adds a list of story requests to an aggregation.
+func (a *Aggregation) Stories(projectID int, storyIDs []int) *Aggregation {
+	for _, storyID := range storyIDs {
+		a.Story(projectID, storyID)
+	}
+	return a
+}
+
+// CommentsOfStory adds a request for getting the comments of a story.
+func (a *Aggregation) CommentsOfStory(projectID, storyID int) {
+	a.requests = append(a.requests, buildCommentsURL(projectID, storyID))
+}
+
+// CommentsOfStories adds multiple requests for getting the comments of a list of stories.
+func (a *Aggregation) CommentsOfStories(projectID int, storyIDs []int) *Aggregation {
+	for _, storyID := range storyIDs {
+		a.CommentsOfStory(projectID, storyID)
+	}
+	return a
+}
+
+// ReviewsOfStory adds multiple requests for getting the reviews of a story.
+func (a *Aggregation) ReviewsOfStory(projectID, storyID int) *Aggregation {
+	a.requests = append(a.requests, buildReviewsURL(projectID, storyID))
+	return a
+}
+
+// ReviewsOfStories adds multiple requests for getting the reviews of multiple stories.
+func (a *Aggregation) ReviewsOfStories(projectID int, storyIDs []int) *Aggregation {
+	for _, storyID := range storyIDs {
+		a.ReviewsOfStory(projectID, storyID)
+	}
+	return a
+}
+
+func buildStoryURLOnlyUsingStoryID(storyID int) string {
+	return fmt.Sprintf("/services/v5/stories/%d", storyID)
+}
+
+func buildStoryURL(projectID, storyID int) string {
+	return fmt.Sprintf("/services/v5/projects/%d/stories/%d", projectID, storyID)
+}
+
+func buildCommentsURL(projectID, storyID int) string {
+	return fmt.Sprintf("/services/v5/projects/%d/stories/%d/comments", projectID, storyID)
+}
+
+func buildReviewsURL(projectID, storyID int) string {
+	return fmt.Sprintf("/services/v5/projects/%d/stories/%d/reviews?fields=id,story_id,review_type,review_type_id,reviewer_id,status,created_at,updated_at,kind", projectID, storyID)
+}
+
+func maxPagesPagination(total int, perPage int) int {
+	pagesNumber := math.Ceil(float64(total) / float64(perPage))
+	return int(pagesNumber)
+}
+
+func paginate(reqs []string, currentPage, total int, perPage int) []string {
+	firstEntry := (currentPage - 1) * perPage
+	lastEntry := firstEntry + perPage
+
+	if lastEntry > total {
+		lastEntry = total
+	}
+
+	return reqs[firstEntry:lastEntry]
+}
+
+// Send completes the aggregation and sends it to Pivotal Tracker.
+func (a *Aggregation) Send() (*Aggregation, error) {
```

Comment:
I think my previous comments were lost on a rebase or something, but this should return `*http.Response` as the second return value to match the rest of the API in this module.

Reply:
Ah, ok. Maybe just ignore this comment then. We don't need it ourselves. If upstream ever shows a willingness to merge this, we can deal with the concern then.

Reply:
I send bulks of 15 requests in `Send`. So, there are multiple `http.Responses`. 

In my next commit, I've changed `Send` to only send the next bulk in the aggregation and return the corresponding `http.Response`. And then I've added another `SendAll` method, which sends all the bulks in one go. `SendAll` returns the `http.Response`s in a slice. Do you think that is fine?